### PR TITLE
Randomize the output directory to avoid conflicts

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: qa/common/util.sh [hunk 1]
```diff
@@ -244,3 +244,14 @@ function find_available_ports() {
...
+# We need to randomize the output directory so that in case the CI containers
+# end up on the same host, it does not create a conflict
+function get_output_directory() {
+    random_number=$((10000 + $RANDOM % 10000))
+    while [ `ls /tmp/output/ | grep $random_number` ]; do
+        random_number=$(echo "$random_number + 1" | bc)
+    done
+
+    echo /tmp/output/$random_number
+}
```
Comment (aramesh7) [inline qa/common/util.sh:257]: New line

**Review by aramesh7 — CHANGES_REQUESTED** [review]

File: qa/common/util.sh (hunk 1 above)
Comment (Tabrizian) [reply qa/common/util.sh]: Fixed

**Review by aramesh7 — DISMISSED** [review]

File: qa/L0_analyze/test.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ TRITON_SERVER_VERSION="21.02-py3"
 CLIENT_PROTOCOL="grpc"
 PORTS=(`find_available_ports 3`)
 GPUS=(`get_all_gpus_uuids`)
-OUTPUT_MODEL_REPOSITORY=${OUTPUT_MODEL_REPOSITORY:='/tmp/output/model_repository'}
```
Comment (aramesh7) [inline qa/L0_analyze/test.sh:37]: The reason I added this was if we wanted to manually specify an output model repository at the command line during a local test.  Could we make it 
```
OUTPUT_MODEL_REPOSITORY
=${OUTPUT_MODEL_REPOSITORY:=(`get_output_directory`)}
```

File: qa/L0_config/test.sh [hunk 3]
```diff
@@ -32,7 +32,7 @@ TRITON_SERVER_VERSION="21.02-py3"
 CLIENT_PROTOCOL="grpc"
 PORTS=(`find_available_ports 3`)
 GPUS=(`get_all_gpus_uuids`)
-OUTPUT_MODEL_REPOSITORY=${OUTPUT_MODEL_REPOSITORY:='/tmp/output/model_repository'}
```
Comment (aramesh7) [inline qa/L0_config/test.sh:35]: Same here

File: qa/L0_export_metrics/test.sh [hunk 4]
```diff
@@ -32,7 +32,7 @@ TRITON_LAUNCH_MODE="local"
 CLIENT_PROTOCOL="http"
 PORTS=(`find_available_ports 3`)
 GPUS=(`get_all_gpus_uuids`)
-OUTPUT_MODEL_REPOSITORY=${OUTPUT_MODEL_REPOSITORY:='/tmp/output/model_repository'}
```
Comment (aramesh7) [inline qa/L0_export_metrics/test.sh:35]: And here

File: qa/L0_server_launch_modes/test.sh [hunk 5]
```diff
@@ -15,7 +15,7 @@
 source ../common/util.sh
 
 rm -f *.log
-OUTPUT_MODEL_REPOSITORY='/tmp/output/model_repository'
```
Comment (aramesh7) [inline qa/L0_server_launch_modes/test.sh:18]: Maybe good to add that here as well

**Review by aramesh7 — CHANGES_REQUESTED** [review]

File: qa/L0_analyze/test.sh (hunk 2 above)
Comment (Tabrizian) [reply qa/L0_analyze/test.sh:37]: I see. Sorry. It's fixed now.

**Review by aramesh7 — APPROVED** [review]

**Pull request merged**